rasterio/_io.pyx

```diff
@@ -1458,11 +1458,16 @@ cdef class DatasetWriterBase(DatasetReaderBase):
 
         band = self.band(1)
 
+        if not all(MaskFlags.per_dataset in flags for flags in self.mask_flag_enums):
+            try:
+                exc_wrap_int(GDALCreateMaskBand(band, MaskFlags.per_dataset))
```

Comment:
The use of `band` here is confusing since it's creating a mask for the dataset. But maybe that's just the GDAL API?

Could we use [`GDALCreateDatasetMaskBand`](http://www.gdal.org/gdal_8h.html#a7a08585a35285b7bb7398afe964a3c91) instead?

Reply:
I don't see much documentation for that method @perrygeo , so I think we should avoid it.